deployments/tools/export/lakefs_export.py

```diff
@@ -0,0 +1,117 @@
+import subprocess
+import argparse
+import os
+import shlex
+import time
+from datetime import datetime
+from string import Template
+
+LAKEFS_ACCESS_KEY = os.getenv('LAKEFS_ACCESS_KEY')
+LAKEFS_SECRET_KEY = os.getenv('LAKEFS_SECRET_KEY')
+LAKEFS_ENDPOINT = os.getenv('LAKEFS_ENDPOINT')
+S3_ACCESS_KEY = os.getenv('S3_ACCESS_KEY')
+S3_SECRET_KEY = os.getenv('S3_SECRET_KEY')
+
+SUCCESS_MSG = "Export completed successfully!"
+
+
+def create_rclone_conf_file():
+    with open('rclone.conf.template') as f:
+        src = Template(f.read())
+        vars = {'lakefs_access_key': LAKEFS_ACCESS_KEY, 'lakefs_secret_key': LAKEFS_SECRET_KEY, 'lakefs_endpoint': LAKEFS_ENDPOINT, 's3_access_key': S3_ACCESS_KEY, 's3_secret_key': S3_SECRET_KEY}
+        res = src.substitute(vars)
+    with open('rclone.conf', 'w+') as f:
+        f.write(res)
+    return
+
+
+def set_args():
+    parser = argparse.ArgumentParser(description='Process lakeFS export.')
+
+    parser.add_argument('Repo', metavar='repo', type=str, help='the name of lakeFS repository')
+    parser.add_argument('Dest', metavar='dest', type=str, help='path of destination')
+    parser.add_argument('--branch', metavar='branch', type=str, action='store', help='the relevant branch in the repository to export from')
+    parser.add_argument('--commit_id', metavar='commit', type=str, action='store', help='the relevant commit on the repository to export from')
+    parser.add_argument('--prev_commit_id', metavar='previous-commit', type=str, action='store', help='a previous commit reference to export only the diff between it and the head of the branch')
+
+    args = parser.parse_args()
+    return args
+
+
+def write_status_file(success, status_file_name):
+    if success:
+        with open(status_file_name,'w') as f:
+            f.write(SUCCESS_MSG)
+    else:
+        with open('errors','r') as errors_file, open(status_file_name,'w') as status_file:
+            for line in errors_file:
+                if line.startswith('-'):
+                    status_file.write("path was missing on the source: " + line)
+                elif line.startswith('+'):
+                    status_file.write("path was missing on the destination: " + line)
+                elif line.startswith('*'):
+                    status_file.write("path was present in source and destination but different: " + line)
+                elif line.startswith('!'):
+                    status_file.write("there was an error reading or hashing the source or dest: " + line)
+    return
+
+
+def main():
+
+    # create rclone configuration file
+    create_rclone_conf_file()
```

Comment:
Why do we need to create a file?  That can fail.  Why not just set [environment variables](https://rclone.org/docs/#environment-variables) for rclone?